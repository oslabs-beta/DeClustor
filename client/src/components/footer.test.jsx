import React from 'react';
import { render } from '@testing-library/react';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import Footer from './Footer';

test('renders Footer component with correct styles and content', () => {
  const theme = createTheme({
    palette: {
      primary: {
        main: '#3f51b5', // Ensure the correct format for primary color
      },
      secondary: {
        main: '#f50057', // Ensure the correct format for secondary color
      },
    },
  });

  const { asFragment, getByText } = render(
    <ThemeProvider theme={theme}>
      <Footer />
    </ThemeProvider>
  );

  // Check if the Footer contains the correct text
  expect(getByText(/DeClustor Â©/)).toBeInTheDocument();

  // Use snapshot testing to check rendered output
  expect(asFragment()).toMatchSnapshot();
});
