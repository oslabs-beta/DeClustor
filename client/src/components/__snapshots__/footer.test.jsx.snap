// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer component with correct styles and content 1`] = `
<DocumentFragment>
  <footer
    class="MuiBox-root css-ymuci6"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        DeClustor Â© 2024
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
